### DISCUSSION TITLE
removing or archiving old data, rolling data files

### DISCUSSION THREAD

**Discussion by meonkeys** [GothenburgBitFactory/taskwarrior#2972] [General]
I'm curious about the Taskwarrior data files `backlog.data`, `completed.data`, `pending.data`, and `undo.data`. Are these "permanent" (must be left as-is), or can these be "rolled" like logfiles? I rolled the `undo.data` file by renaming and compressing the current one and a new `undo.data` file was automatically created. I forget where I got the idea that it was OK to do this.

Is there some way to easily save space by rolling, archiving, or even deleting old data? Sorry if I missed this in the docs or a mailing list/github discussion.

**Comment by djmitche** [comment]
They're all a little different, despite having the same extension:

* `pending.data` contains only pending tasks, and, assuming you keep your pending tasks at a relatively constant count, stays a reasonable size.
* `completed.data` contains all non-pending tasks, and grows without bound.  It can be deleted, if you're OK with losing all that history.  Deleting may interact in "interesting" ways with sync.
* `backlog.data` contains changes not yet transmitted to `taskd` by `task sync`.  If you sync regularly, it should always be quite small.
* `undo.data` contains all changes made to all tasks.  It is used to support `task undo` but also to generate the history shown in `task info` and a few other places.  It grows without bound.  It can be safely deleted, if you're OK not having that history in `task info`.

None can really be "rolled" like a logfile -- they're not log files.  I don't see much point to saving the file contents elsewhere if you're going to delete the files Taskwarrior uses.

Taskchampion uses a completely different model, by the way: a database of all tasks keyed by uuid, with a separate list of un-sync'd operations.  `task undo` is supported by that list of un-sync'd operations.  It doesn't yet support task history (that's #2928).

**Reply by chaoticlonghair** [reply]
So, `backlog.data` can be safely deleted if I don't use `taskd` or `task sync`.

Do I understand correctly?
